flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/PushPartitionIntoTableSourceScanRule.scala

```diff
@@ -60,20 +67,26 @@ class PushPartitionIntoTableSourceScanRule extends RelOptRule(
   override def onMatch(call: RelOptRuleCall): Unit = {
     val filter: Filter = call.rel(0)
     val scan: LogicalTableScan = call.rel(1)
+    val context = call.getPlanner.getContext.unwrap(classOf[FlinkContext])
```

Comment:
we need some tests for this rule

Reply:
We already have `PushPartitionIntoTableSourceScanRuleTest`.

Reply:
but it doesn't cover your stats handle logics

Reply:
I think `CatalogStatisticsTest` is more professional to test stats.

Reply:
So who will cover the correctness about the stats after partition pruning?

Reply:
`CatalogStatisticsTest` will trigger partition pruning.

Reply:
ok, got it